applications: nrf_desktop: Improve memory layout documentation (#13254)
nrfconnect/sdk-nrf: applications/nrf_desktop/README.rst

.. _nrf_desktop_flash_memory_layout:

Memory layout
=============

You must define the memory layout as a part of the application configuration for a given board.
The set of required partitions differs depending on configuration:

* There must be at least one partition where the code is stored.
* There must be one partition for storing :ref:`zephyr:settings_api`.

> * The bootloader, if enabled, adds more partitions to the set.

Reviewer: ```suggestion
* If the bootloader is enabled, it adds more partitions to the set.
```

* When using an SoC with multiple cores, the firmware for additional cores will add additional partitions to the set.
  For example, nRF53 SoC uses ``HCI RPMsg`` firmware image for the network core to use the core for Bluetooth LE communication.

.. important::
   Before updating the firmware, make sure that the data stored in the settings partition is compatible with the new firmware.
   If it is incompatible, erase the settings area before using the new firmware.

The memory layout is defined through one of the following methods:

* `Memory layout in DTS`_
* `Memory layout in Partition Manager`_

By default, a Zephyr-based application defines memory layout in the DTS.
If enabled, the :ref:`partition_manager` defines a new memory layout that is used instead of the memory layout defined in the DTS.

> The :kconfig:option:`CONFIG_PARTITION_MANAGER_ENABLED` Kconfig option value can be used to check if the Partition Manager is enabled in the current build.

Author: Already applied as fix from @peknis

> The option is automatically selected in case :ref:`ug_multi_image` feature is used to build the application together with a bootloader or nRF53 network core image.

Reviewer 2: What is the nRF53 network core image? nRF53 network core has many potential images that you can use. Let's make this sentence more generic and less target specific

```suggestion
The option is automatically selected as part of the :ref:`ug_multi_image` feature to build the application with more than one image.
```

Author: Ok, we can update it.

> The option is automatically selected in case :ref:`ug_multi_image` feature is used to build the application together with a bootloader or nRF53 network core image.

Author: Slightly rephrased following @kapi-no's suggestion

Enabling the :ref:`nrf_desktop_bluetooth_guide_fast_pair` also results in using the Partition Manager.
To store the Fast Pair Provisioning data, the Fast Pair integration in the |NCS| uses partition defined by the Partition Manager.

Memory layout in DTS
--------------------

In case you rely on a non-volatile memory layout described in DTS files, define the ``partitions`` child node under the DTS node that represents the non-volatile memory.

> For example, the nRF52 Series use non-volatile flash memory represented by the flash device node (``&flash0``).

Reviewer 3: ```suggestion
For example, the nRF52 Series devices use non-volatile flash memory represented by the flash device node (``&flash0``).
```

Make sure to also update the DTS chosen nodes representing the code partition (``zephyr,code-partition``) and flash (``zephyr,flash``) if needed.

> If you wish to change the default memory layout of the board without editing board-specific files, edit the DTS overlay file.

Reviewer 3: ```suggestion
If you wish to change the default memory layout of the board without editing the board-specific files, edit the DTS overlay file.
```
